gwcs/wcs.py

```diff
@@ -634,3 +641,211 @@ def fix_inputs(self, fixed):
+        if degree is None and max_error is None and max_rms is None:
+            raise ValueError(
+                "max_error must be specified if degree is None")
+        fits_dict = {}
+        transform = self.forward_transform
+        # Determine reference points.
+        (xmin, xmax), (ymin, ymax) = bounding_box
+        crpix1 = int((xmax - xmin) / 2)
+        crpix2 = int((ymax - ymin) / 2)
+        crval1, crval2 = transform(crpix1, crpix2)
+        hdr = fits.Header()
+        hdr['naxis'] = 2
+        hdr['naxis1'] = xmax
+        hdr['naxis2'] = ymax
+        hdr['ctype1'] = 'RA---TAN-SIP'
+        hdr['ctype2'] = 'DEC--TAN-SIP'
+        hdr['CRPIX1'] = crpix1 + 1
+        hdr['CRPIX2'] = crpix2 + 1
+        hdr['CRVAL1'] = crval1
+        hdr['CRVAL2'] = crval2
+        hdr['cd1_1'] = 0  # Placeholders
+        hdr['cd1_2'] = 0
+        hdr['cd2_1'] = 0
+        hdr['cd2_2'] = 0
+        # Now rotate to native system and deproject
+        ntransform = (transform
+                      | RotateCelestial2Native(crval1, crval2, 180)
+                      | Sky2Pix_TAN())
+        # Evaluate this transform on all pixels in bounding box region
+        y, x = np.mgrid[ymin:ymax, xmin:xmax]  # Check on the endpoints needing to be 1 larger
+        u = x - crpix1
+        v = y - crpix2
+        undist_x, undist_y = ntransform(x, y)
+        # The fitting section.
+        # Fit the forward case.
+        fit_poly_x, fit_poly_y, max_resid = _fit_2D_poly(degree, max_error,
+                                                             u, v, undist_x, undist_y,
+                                                             verbose=verbose)
+        cdmat = np.array([[fit_poly_x.c1_0.value, fit_poly_x.c0_1.value],
+                          [fit_poly_y.c1_0.value, fit_poly_y.c0_1.value]])
+        det = cdmat[0, 0] * cdmat[1, 1] - cdmat[0, 1] * cdmat[1, 0]
+        U = (cdmat[1, 1] * undist_x - cdmat[0, 1] * undist_y) / det
+        V = (-cdmat[1, 0] * undist_x + cdmat[0, 0] * undist_y) / det
+        fit_inv_poly_u, fit_inv_poly_v, max_inv_resid = _fit_2D_poly(degree,
+                                                            max_inv_error,
+                                                            U, V, u - U, v - V,
+                                                            verbose=verbose)
+        pdegree = fit_poly_x.degree
+        if pdegree > 1:
+            hdr['a_order'] = pdegree
+            hdr['b_order'] = pdegree
+            ipdegree = fit_inv_poly_u.degree
+            hdr['ap_order'] = ipdegree
+            hdr['bp_order'] = ipdegree
+            cd, sip_poly_x, sip_poly_y = _reform_poly_coefficients(fit_poly_x, fit_poly_y)
+            _store_2D_coefficients(hdr, sip_poly_x, 'A')
+            _store_2D_coefficients(hdr, sip_poly_y, 'B')
+            _store_2D_coefficients(hdr, fit_inv_poly_u, 'AP', keeplinear=True)
+            _store_2D_coefficients(hdr, fit_inv_poly_v, 'BP', keeplinear=True)
```

Comment:
I think these 5 function calls each use almost the same double-loop:
```python
for i in range(0, degree + 1):
    for j in range(0, degree + 1):
        if (i + j > 1) and (i + j < degree + 1):
```
I feel there should be a way of combining all these header updates into a single loop.